### ISSUE TITLE
Nano: Automatically rescale learning rate for multi-instance training

### ISSUE THREAD

**Issue by yangw1234** [intel-analytics/BigDL#4855]
pytorch reference implementation

https://github.com/Lightning-AI/lightning/blob/1.6.4/pytorch_lightning/strategies/horovod.py#L103-L113

**Comment by jason-dai** [comment]
I think we probably need both linear scaling and warmup

**Comment by yangw1234** [comment]
> I think we probably need both linear scaling and warmup
Sure

**Comment by TheaperDeng** [comment]
Just for reference (code in Chronos might be affected):
Since we will automatically set `num_processes` for our users according to their core nums. So we will change the batch_size for each executor to make sure the result is the same on different platform.

https://github.com/intel-analytics/BigDL/blob/main/python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py#L284-L292
